test/infrastructure/docker/docker/util.go

```diff
@@ -102,34 +105,38 @@ func List(filters ...string) ([]*types.Node, error) {
 }
 
 func list(visit func(string, *types.Node), filters ...string) error {
-	args := []string{
-		"ps",
-		"-q",         // quiet output for parsing
-		"-a",         // show stopped nodes
-		"--no-trunc", // don't truncate
-		// filter for nodes with the cluster label
-		"--filter", "label=" + clusterLabelKey,
-		// format to include friendly name and the cluster name
-		"--format", fmt.Sprintf(`{{.Names}}\t{{.Label "%s"}}\t{{.Image}}\t{{.Status}}`, clusterLabelKey),
+	ctx := context.Background()
+	cli, err := dockerClient.NewClientWithOpts(dockerClient.FromEnv, dockerClient.WithAPIVersionNegotiation())
+	if err != nil {
+		return errors.Wrap(err, "failed to connect to docker daemon")
+	}
+
+	options := dockerTypes.ContainerListOptions{
+		All:     true,
+		Limit:   -1,
+		Filters: dockerFilters.NewArgs(dockerFilters.Arg("label", clusterLabelKey)),
```

Comment:
Side note: I think that we should drift away from kind label names ("io.x-k8s.kind.cluster", "io.x-k8s.kind.role") and start using CAPD dedicated ones

Reply:
Should we get an enhancement issue filed to track this? Should be easy to do. I'm just not sure of what we would need to do to handle backwards compatibility.

Reply:
It is just changing the labels names; there is no need to handle backward compatibility because CAPD cluster are ephemeral / for dev pouposes  only/ they do not survive restart

Reply:
Oh, very good point! :)